tests/basic/telemetry_test.go

```diff
@@ -233,77 +233,93 @@ var _ = Describe("{ProfileOnlyDiags}", func() {
 	var existingDiags string
 	var newDiags string
 	var err error
+	var pxInstalled bool
+	var skipTest = false
 
 	It("has to collect, validate profile diags on S3", func() {
 		contexts = make([]*scheduler.Context, 0)
 		// One node at a time, collect diags and verify in S3
 		for _, currNode := range node.GetWorkerNodes() {
-			// Get the most recent profile diags for comparison
-			Step(fmt.Sprintf("Check latest profile diags on node %v", currNode.Name), func() {
-				logrus.Infof(" Getting latest profile  diags on %v", currNode.Name)
-				existingDiags, err = telemetryRunCmd(fmt.Sprintf("ls -t /var/cores/%s-*.{stack,heap}.gz | head -n 2", currNode.Name),
-					currNode, nil)
-				if err == nil {
-					logrus.Infof("Found latest profiles diags on node %s:\n%s ", currNode.Name, existingDiags)
-				} else {
-					existingDiags = ""
-				}
-			})
-			// Issue a profile diags to generate new files.
-			Step(fmt.Sprintf("collect profile diags on node: %s | %s", currNode.Name, currNode.Type), func() {
-				config := &torpedovolume.DiagRequestConfig{
-					DockerHost: "unix:///var/run/docker.sock",
-					Profile:    true,
-				}
-				if !TelemetryEnabled(currNode) { // Not sure why this is done?
-					logrus.Debugf("Telemetry not enabled, sleeping for 5 mins")
-					time.Sleep(5 * time.Minute)
-				}
-				err := Inst().V.CollectDiags(currNode, config, torpedovolume.DiagOps{Validate: true})
-				if err != nil {
-					logrus.Errorf("Failed to collect diags: %v", err)
-				}
-				Expect(err).NotTo(HaveOccurred(), "Profile only diags collected successfully")
-			})
-			// Get the latest files in the directory.  The newly generated files will not equal the most recent. So you know they are new.
-			Step(fmt.Sprintf("Get the new profile diags on node %v", currNode.Name), func() {
-				logrus.Infof("Getting latest profile diags on %66v", currNode.Name)
-				Eventually(func() bool {
-					newDiags, err = telemetryRunCmd(fmt.Sprintf("ls -t /var/cores/%s-*.{stack,heap}.gz | head -n 2", currNode.Name),
+			pxInstalled, err = Inst().V.IsPxInstalled(currNode)
+			if err != nil {
+				logrus.Debugf("Could not get PX status on %s", currNode.Name)
```

Comment:
You dont want to fail if you are unable to get the pxInstalled status ? What are the consequences, check once..